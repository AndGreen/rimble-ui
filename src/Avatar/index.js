import React from 'react'
import styled from 'styled-components'
import { themeGet } from 'styled-system'

import theme from '../theme'

import Box from '../Box'

const size = (props) => {
  switch (props.size) {
    case 'small':
      return `
        height: ${props.theme.space[3]}px;
        width: ${props.theme.space[3]}px;
      `
    case 'medium':
      return `
        height: ${props.theme.space[4]}px;
        width: ${props.theme.space[4]}px;
      `
    case 'large':
      return `
        height: ${props.theme.space[5]}px;
        width: ${props.theme.space[5]}px;
      `
    default:
      return ``
  }
}

const Avatar = styled(Box)`
  & {
    position: relative;
    overflow: hidden;
    border-radius: 100%;
    background: center no-repeat;
    background-size: cover;
    background-color: ${ props => props.theme.colors.grey };
    background-image: url(${props => props.src});
  }

  ${size}
`

Avatar.defaultProps = {
  theme: theme,
  ratio: 1,
  size: '2rem'
}

Avatar.displayName = 'Avatar'

export default Avatar
