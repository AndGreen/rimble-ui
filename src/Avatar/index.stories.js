import { storiesOf } from '@storybook/react'

import React from 'react'
import Avatar from '../Avatar'

storiesOf('Avatar', module)
  .add('Default', () => <Avatar />)
