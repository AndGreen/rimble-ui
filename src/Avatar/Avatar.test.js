import Avatar from './'

describe('Avatar component sanity', () => {
  it('has name', () => {
    expect(Avatar.displayName).toBe('Avatar')
  })

  it('matches default snapshot', () => {
    const component = render(<Avatar />)
    expect(component).toMatchSnapshot()
  })
})
