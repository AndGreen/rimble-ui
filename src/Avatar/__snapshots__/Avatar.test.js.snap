// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component sanity matches default snapshot 1`] = `
.c0 {
  background-color: #CCC;
  width: 2rem;
  height: 2rem;
}

.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  overflow: hidden;
  border-radius: 100%;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url(test.jpg);
}

<div
  alt="Avatar"
  class="c0 c1"
  src="test.jpg"
/>
`;

exports[`Avatar component sanity matches themed snapshot 1`] = `
.c1 {
  background-color: #CCC;
  width: 2rem;
  height: 2rem;
}

.c1 {
  box-sizing: border-box;
}

.c2 {
  position: relative;
  overflow: hidden;
  border-radius: 100%;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url(test.jpg);
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <div
    alt="Avatar"
    class="c1 c2"
    src="test.jpg"
  />
</div>
`;
